server-side/managers/sharingManager.js

```diff
@@ -0,0 +1,38 @@
+import { getInvitationById_service } from '../services/invitation.service.js';
+import { getUserById_service } from '../services/user.service.js';
+import { getProfileById_service, createProfile_service} from '../services/profile.service.js';
+
+export const updateProfiles = async (invitationID) => {
+    const invitation = await getInvitationById_service(invitationID);
+    if (!invitation) {
+        throw new Error('Invitation not found');
+    }
+    const { shareLevel, profileID, invitedUserID } = invitation;
+    const user = await getUserById_service(invitedUserID);
+    if (shareLevel === 'duplicate') {
+        const profileToDuplicate = await getProfileById_service(profileID);
+        const newProfile = await createProfile_service({
+            profileName: profileToDuplicate.profileName,
+            statusBlockedSites: profileToDuplicate.statusBlockedSites,
+            listWebsites: profileToDuplicate.listWebsites,
+        });
+        await newProfile.save();
+        user.profiles.push(newProfile._id);
+        await user.save();
+        return newProfile;
+    } else if (shareLevel === 'view') {
+        if (!user.viewProfiles.includes(profileID)) {
```

Comment:
in case this profileId was already shared with this user you should return an error to the client and show the message in a notification in the UI, so the user will be noticed that this profile was already shared.
the same for view & edit